src/main/java/seedu/address/logic/commands/EditDoctorCommand.java

```diff
@@ -0,0 +1,191 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NRIC;
+import static seedu.address.model.NewModel.PREDICATE_SHOW_ALL_DOCTORS;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.CollectionUtil;
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.NewModel;
+import seedu.address.model.person.Name;
+import seedu.address.model.person.Nric;
+import seedu.address.model.person.doctor.Doctor;
+
+/**
+ * Edits the details of an existing doctor in the address book.
+ */
+public class EditDoctorCommand extends NewCommand {
+
+    public static final String COMMAND_WORD = "edit_d";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the details of the doctor identified "
+            + "by the index number used in the displayed doctor list. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "[" + PREFIX_NAME + "NAME] "
+            + "[" + PREFIX_NRIC + "NRIC]...\n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_NAME + "BOBE "
+            + PREFIX_NRIC + "T5012345A";
+
+    public static final String MESSAGE_EDIT_DOCTOR_SUCCESS = "Edited Doctor: %1$s";
+    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
+    public static final String MESSAGE_DUPLICATE_DOCTOR = "This doctor already exists in the address book.";
+
+    private final Index index;
+    private final EditDoctorDescriptor editDoctorDescriptor;
+
+    /**
+     * @param index of the doctor in the filtered doctor list to edit
+     * @param editDoctorDescriptor details to edit the doctor with
+     */
+    public EditDoctorCommand(Index index, EditDoctorDescriptor editDoctorDescriptor) {
+        requireNonNull(index);
+        requireNonNull(editDoctorDescriptor);
+
+        this.index = index;
+        this.editDoctorDescriptor = new EditDoctorDescriptor(editDoctorDescriptor);
+    }
+
+    @Override
+    public CommandResult execute(NewModel model) throws CommandException {
+        requireNonNull(model);
+        List<Doctor> lastShownList = model.getFilteredDoctorList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_DOCTOR_DISPLAYED_INDEX);
+        }
+
+        Doctor doctorToEdit = lastShownList.get(index.getZeroBased());
+        Doctor editedDoctor = createEditedDoctor(doctorToEdit, editDoctorDescriptor);
+
+        if (!doctorToEdit.isSame(editedDoctor) && model.hasDoctor(editedDoctor)) {
```

Comment:
I think it just refers to
1. Edited Doctor is the Same with Edited Doctor (By Checking NRIC) .
2. **If it is not** and model has **a different Doctor in database that is same with edited doctor**, it will just calls a duplicated doctor message.

Reply:
oh ok my mistake